build.fsx

```diff
@@ -99,38 +286,81 @@ Target.create "AssemblyInfo" (fun _ ->
              then [AssemblyInfo.InternalsVisibleTo("FsCheck.Test")] else []))
     )
 )
-
-// --------------------------------------------------------------------------------------
-// Clean build results
-
-Target.create "Clean" (fun _ ->
-    Shell.cleanDirs ["bin"; "temp"; "output"]
-)
+*)
 
 // --------------------------------------------------------------------------------------
 // Build library & test project
 
+type HaveBuilt = HaveBuilt
+let build (_ : HaveCleaned) : HaveBuilt =
+    Console.Write "Performing dotnet restore... "
+    runProcess "dotnet" ["restore" ; solution]
+    Console.WriteLine "done."
+    Console.Write "Performing dotnet build... "
+    runProcess "dotnet" ["build" ; solution ; "--configuration" ; "Release"]
+    Console.WriteLine "done."
+    HaveBuilt
+
+(*
 Target.create "Build" (fun _ ->
     DotNet.restore id solution
     DotNet.build (fun opt -> { opt with Configuration = DotNet.BuildConfiguration.Release }) solution
 )
+*)
 
 // --------------------------------------------------------------------------------------
 // Run the unit tests
 
-let runAndCheck (f: unit -> ProcessResult) =
-  let result = f()
-  if not result.OK then
-    failwithf "%A" result
+type HaveTested = HaveTested
+let runDotnetTest (_ : HaveCleaned) : HaveTested =
+    Console.Write "Performing dotnet test... "
+    runProcess "dotnet" ["test" ; "tests/FsCheck.Test" ; "--configuration" ; "Release"]
+    Console.WriteLine "done."
+    HaveTested
 
+(*
 Target.create "RunTests" (fun _ ->
   "tests/FsCheck.Test/"
   |> DotNet.test (fun opt -> { opt with Configuration = DotNet.BuildConfiguration.Release })
 )
+*)
 
 // --------------------------------------------------------------------------------------
 // Build a NuGet package
 
+type HavePacked = HavePacked
+let packNuGet (_ : HaveTested) : HavePacked =
+    Console.Write "Performing dotnet pack... "
+    let releaseNotes =
+        releaseNotes.Notes
+        |> String.concat "\n"
+    runProcess "dotnet" ["pack" ; "--configuration" ; "Release" ; $"-p:Version=%s{buildVersion}" ; "--output" ; "bin" ; $"-p:PackageReleaseNotes=%s{releaseNotes}"]
+
+    // I *believe* it is impossible to have a fixed (not floating) version number from a source reference
```

Comment:
*sob* I spent *hours* trying to find a better way than "crack open the nuspec and fix it up manually".

Reply:
Oh no. I'm sorry. This is amazing attention to detail though. I would've probably said it's ok to let it float...